frontends/web/src/components/badge/badge.tsx

```diff
@@ -14,22 +14,31 @@
  * limitations under the License.
  */
 
-import { ReactNode } from 'react';
+import { ReactElement, ReactNode } from 'react';
 import style from './badge.module.css';
 
 type TBadgeStyles = 'success' | 'warning'; // TODO: not yet implemented 'info' | 'danger'
 
 type TProps = {
-  children: ReactNode;
+  children?: ReactNode;
   type?: TBadgeStyles;
-}
+  icon?: (props: { className: string }) => ReactElement;
+} & JSX.IntrinsicElements['span'];
 
 export const Badge = ({
   children,
+  className,
+  icon,
   type = 'success',
+  ...props
 }: TProps) => {
+  const withChildrenStyle = children !== undefined ? style.withChildren : '';
+  const iconOnlyStyle = (children === undefined && icon) ? style.iconOnly : '';
   return (
-    <span className={`${style.badge} ${style[type]}`}>
+    <span
+      className={`${style.badge} ${style[type]} ${withChildrenStyle} ${iconOnlyStyle} ${className}`}
+      {...props}>
+      {icon && icon({ className: style.badgeIcon })}
```

Comment:
If it's not too difficult, could we add a tiny space like 4px between the icon & text (when theres one)?

IMO looks a bit neater with it:

with extra space:

![Screenshot 2023-12-12 at 14 04 54](https://github.com/digitalbitbox/bitbox-wallet-app/assets/28676406/71398b1e-77fa-4aa0-9b5e-54cbeee128bd)

without:

<img width="206" alt="Screenshot 2023-12-12 at 14 04 16" src="https://github.com/digitalbitbox/bitbox-wallet-app/assets/28676406/660256b9-6f62-46d0-a654-68f39e3283d2">

Reply:
> IMO looks a bit neater with

It has a small space, the main reason I didn't want to add margin-left is that in the case when there is a long name and the icon breaks on a 2nd line... but now that we don't have the "Account" aanymore I think it makes sense to add it again. I will update the PR.

Reply:
tysm!

Reply:
![Screen Shot 2023-12-12 at 14 59 48](https://github.com/digitalbitbox/bitbox-wallet-app/assets/546900/b6479ac2-6f81-421d-b2ea-8e74fac21424)

and in cases where it shows root fingerprint

![Screen Shot 2023-12-12 at 15 00 11](https://github.com/digitalbitbox/bitbox-wallet-app/assets/546900/d4e59186-b1ae-483b-a7cd-8e29e3d416ab)